packages/playwright-core/src/server/page.ts

```diff
@@ -455,7 +456,31 @@ export class Page extends SdkObject {
     this._locatorHandlers.delete(uid);
   }
 
-  async performLocatorHandlersCheckpoint(progress: Progress) {
+  async performActionPreChecks(progress: Progress) {
+    await this._performWaitForNavigationCheck(progress);
+    progress.throwIfAborted();
+    await this._performLocatorHandlersCheckpoint(progress);
```

Comment:
What if this causes navigation?

Reply:
Good call, I'll wait for navigations after this one as well.